manifests/master/generate_types.pp

```diff
@@ -0,0 +1,66 @@
+# Use ``incrond`` to run ``puppet generate types`` when the necessary files
+# have been changed
+#
+# @param enable
+#   Enable ``puppet generate types`` management
+#
+# @param trigger_on_puppetserver_update
+#   Run ``puppet generate types`` on all environments if the ``puppetserver``
+#   application is updated
+#
+# @param trigger_on_puppet_update
+#   Run ``puppet generate types`` on all environments if the ``puppet``
+#   application is updated
+#
+# @param trigger_paths
+#   The paths that should be watched
+#
+#   * Ruby ``Dir`` compatible path globs are supported
+#   * The string ``PUPPET_ENVIRONMENTPATH`` will be subsituted with all known
+#   * Puppet Environment Paths
+#
+class pupmod::master::generate_types (
+  Boolean                     $enable                         = true,
+  Boolean                     $trigger_on_puppetserver_update = true,
+  Boolean                     $trigger_on_puppet_update       = true,
+  Array[Stdlib::AbsolutePath] $trigger_paths                  = pupmod::generate_types_munge([
+                                '/PUPPET_ENVIRONMENTPATH',
+                                '/PUPPET_ENVIRONMENTPATH/*/modules/*/lib/puppet/type/*.rb'
```

Comment:
IN_NO_LOOP would be more effective if it was operating on directories, not individual files.

Reply:
Modified this to handle the `types` directory and to also monitor `modules` in case new modules are added.